ecs-agent/netlib/platform/common_linux.go

```diff
@@ -695,22 +695,31 @@ func (c *common) configureBranchENI(ctx context.Context, netNSPath string, eni *
 		"NetNSPath": netNSPath,
 	})
 
-	var cniNetConf ecscni.PluginConfig
+	c.os.Setenv(IPAMDataPathEnv, filepath.Join(c.stateDBDir, IPAMDataFileName))
+
+	var cniNetConf []ecscni.PluginConfig
 	var err error
 	add := true
 
 	// Generate CNI network configuration based on the ENI's desired state.
 	switch eni.DesiredStatus {
 	case status.NetworkReadyPull:
-		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeVlan)
-	case status.NetworkReady:
-		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeTap)
+		// Setup bridge to connect task network namespace to TMDS running in host's primary netns.
+		if eni.IsPrimary() {
+			cniNetConf = append(cniNetConf, createBridgePluginConfig(netNSPath))
```

Comment:
if we weren't doing this before, how was this working at all?

Reply:
it wasn't working at all. ENI is configured successfully, task starts and has network connectivity, but any sort of api call that required task credentials failed since TMDS was not reachable from the task netns.